src/FitImage.js

```diff
@@ -88,8 +81,9 @@ class FitImage extends Image {
 
   onLoad() {
     this.setState({ isLoading: false });
+    this.refreshStateSize();
```

Comment:
It causes double check(getImage API call). Could you try to use cached image size?